examples/ex0.cpp

```diff
@@ -0,0 +1,96 @@
+//                                MFEM Example 0
+//
+// Compile with: make ex0
+//
+// Sample runs:  ex0
+//               ex0 -m ../data/fischera.mesh
+//               ex0 -m ../data/square-disc.mesh -o 2
+//
+// Description: This example code demonstrates the most basic usage of MFEM
+//              to define a simple finite element discretization of the Laplace
+//              problem -Delta u = 1 with homogeneous Dirichlet boundary
+//              conditions. Besides the mesh, the only available option to
+//              the user is the polynomial order.
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+
+int main(int argc, char *argv[])
+{
+   // 1. Parse command line options
+   const char *mesh_file = "../data/star.mesh";
+   int order = 1;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree)");
+   args.Parse();
+   if (!args.Good())
+   {
+      args.PrintUsage(cout);
+      return 1;
+   }
+   args.PrintOptions(cout);
+
+   // 2. Read the mesh from the given mesh file.
+   Mesh mesh (mesh_file, 1, 1);
+
+   // 3. Define a finite element space on the mesh. Here we use continuous
+   // Lagrange finite elements of the specified order.
+   H1_FECollection fec(order, mesh.Dimension());
+   FiniteElementSpace fespace(&mesh, &fec);
+   cout << "Number of unknowns: " << fespace.GetTrueVSize() << endl;
+
+   // 4. Mark all boundary conditions as essential (Dirichlet)
+   Array<int> ess_tdof_list;
+   if (mesh.bdr_attributes.Size())
+   {
+      Array<int> ess_bdr(mesh.bdr_attributes.Max());
+      ess_bdr = 1;
+      fespace.GetEssentialTrueDofs(ess_bdr, ess_tdof_list);
+   }
```

Comment:
To simplify things further, should we add `FiniteElementSpace::GetEssentialTrueDofs(Array<int> &ess_tdof_list);` that does the above internally and returns the dofs on the whole boundary?

Reply:
I added `FiniteElementSpace::GetBoundaryTrueDofs` that does this. I think that's a bit clearer than adding an overloaded version of `GetEssentialTrueDofs`.